.github/workflows/test_and_release.yml

```diff
@@ -0,0 +1,154 @@
+name: Test and Release
+
+on:
+  workflow_dispatch:
+  release:
+    types:
+      - published
+  pull_request:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+-dev
+
+jobs:
+  test:
+    name: Run Insight API tests
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12'
+
+      - name: Enable NPM cache
+        uses: actions/cache@v2
+        with:
+          path: '~/.npm'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install NPM dependencies
+        run: npm ci
+
+      - name: Run tests
+        run: npm run test
+
+  release-npm:
+    name: Release NPM package
+    runs-on: ubuntu-20.04
+    needs: test
+    if: ${{ github.event_name == 'release' }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check package version matches tag
+        uses: geritol/match-tag-to-package-version@0.1.0
+        env:
+          TAG_PREFIX: refs/tags/v
+
+      - name: Enable NPM cache
+        uses: actions/cache@v2
+        with:
+          path: '~/.npm'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install NPM dependencies
+        run: npm ci
+
+      - name: Set release tag
+        uses: actions/github-script@v3
+        id: tag
+        with:
+          result-encoding: string
+          script: |
+            const tag = context.payload.release.tag_name;
+            const [, major, minor] = tag.match(/^v([0-9]+)\.([0-9]+)/);
+            return (tag.includes('dev') ? `${major}.${minor}-dev` : 'latest');
+
+      - name: Publish NPM package
+        uses: JS-DevTools/npm-publish@v1
+        with:
+          token: ${{ secrets.NPM_TOKEN }}
+          tag: ${{ steps.tag.outputs.result }}
+
+  release-docker:
+    name: Release Docker image
+    runs-on: ubuntu-20.04
+    needs: test
```

Comment:
release npm

Reply:
Why? This should run in parallel to the `release-npm` job. The `build-args` needed in the Dockerfile are [passed here already](https://github.com/dashevo/insight-api/blob/github-actions/.github/workflows/test_and_release.yml#L143).

Reply:
It installs npm package which should be published before. Please take a look at Docketfile.

Reply:
I think I see what you mean now, it is [these lines](https://github.com/dashevo/insight-api/blob/master/docker/Dockerfile#L25) that require the NPM release to be pushed first right? What does `dashcore-node` actually do, it looks like JS implementation of dashd? How is this different to the `dash` package?

Reply:
`dash` package is a light client that uses DAPI to fetch, verify, and broadcast data to Dash Network. `dashcore-node` represents a full node.